### PR TITLE
[Product Block Editor]: introduce `<SectionActions />` component

### PR DESCRIPTION
**Pull request by retrofox**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

This PR introduces a new `<SectionActions />` slot component: an abstraction on top of the `<BlockFill />` component to simplify the API.

### Usage

In a block that is a child of a section block, you can populate the section actions:

```jsx

function CustomProductBlockEdit() {
  return (
    <>
      <SectionActions>
	<Button
		onClick={ handleProductAction }
		variant="secondary"
	>
		{ __( 'Product action!', 'woocommerce' ) }
	</Button>
      </SectionActions>

      // The rest of the block edit component
    </>
  );
}
```

### Relevant changes

* `clientId` prop was removed from BlockFill and BlockSlot components. It's picked by using the core hook
* Add the `<SectionActions>` slot component
* Update the product list block using the new slot to render the `Add products` button
* Janitorial, TS, tidy minor changes.

Closes # .

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Enble the new Product Editor
2. Create/edit a product setting Grouped product type
image

3. Scroll down to see the `Products in this group` section
4. Confirm you see the `Add products` button
5. Confirm it works as expected
image

Optionally
7. Install the React Developer Tools for your browser
8. Open the React Developer Tools tab
9. Filter the components by the `SectionActions` name

image

### Changelog entry

-   [ ] Automatically create a changelog entry from the details below.

<details>

#### Significance

-   [x] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [x] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Comment 

</details>

### CODE REVIEW

File: packages/js/product-editor/src/components/block-slot-fill/block-fill.tsx [hunk 1]
```diff
@@ -34,7 +39,11 @@ export function BlockFill( {
 		[ clientId, slotContainerBlockName ]
 	);
 
-	if ( ! closestAncestorClientId ) return null;
+	if ( ! closestAncestorClientId ) {
+		// eslint-disable-next-line no-console
+		console.warn( 'No closest ancestor client ID found for block fill.' );
```
Comment (louwie17) [inline packages/js/product-editor/src/components/block-slot-fill/block-fill.tsx:44]: Nice!

File: packages/js/product-editor/src/components/block-slot-fill/index.ts [hunk 2]
```diff
@@ -1,3 +1,4 @@
 export * from './block-fill';
 export * from './block-slot';
+export * from './section-actions/index';
```
Comment (louwie17) [inline packages/js/product-editor/src/components/block-slot-fill/index.ts:3]: Not mandatory, but the `/index` is not really necessary.

**Review by louwie17 — APPROVED** [review]
LGTM 🚀 Nice work, tested well!

File: packages/js/product-editor/src/components/block-slot-fill/index.ts (hunk 2 above)
Comment (retrofox) [reply packages/js/product-editor/src/components/block-slot-fill/index.ts:3]: going to remove it. For some reason, it wasn't working without the file name. Going to remove it in a follow up

Comment (retrofox) [reply packages/js/product-editor/src/components/block-slot-fill/index.ts:3]: better late than never: https://github.com/woocommerce/woocommerce/pull/46515/commits/593422e9d7c5d7cb9f295ebd87f491b28c612f11

**Pull request merged**